local-cli/install/install.js

```diff
@@ -10,7 +11,17 @@ log.heading = 'rnpm-install';
 function install(args, config) {
   const name = args[0];
 
-  var res = spawnSync('npm', ['install', name, '--save'], spawnOpts);
+  const projectDir = config.getProjectRoots()[0];
+  const isYarnAvailable =
+    yarn.getYarnVersionIfAvailable() &&
+    yarn.isGlobalCliUsingYarn(projectDir);
+
+  var res;
+  if (isYarnAvailable) {
+    res = spawnSync('yarn', ['add', name], spawnOpts);
+  } else {
+    res = spawnSync('npm', ['install', name, '--save'], spawnOpts);
```

Comment:
First of all, thank you for a really nice feedback ❤️. I think I can help addressing this one:
> 4 - RARE CASE: Caller want to stick with npm
>
> like react-native-cli/index which allows user can init project with --npm flag and it's happen only before react-native have been installed - it doesn't use local-cli during its process. btw I don't think we need to support --npm flag in local-cli right now.

I think it won't be complicated to check if the current repo has been scaffolded using `--npm` flag or not. It can be added to the manager's constructor, right next to the check if user have `yarn` installed. What do you think?